src/js/common/schemaValidatedComponent.js

```diff
@@ -0,0 +1,249 @@
+                                        }
+                                    },
+                                    {
+                                        "properties": {
+                                            "method": { "type": "string", "required": true},
+                                            "this": { "type": "string", "required": true }
+                                        }
+                                    }
+                                ],
+                                "properties": {
+                                    "args": { "type": "array"}
+                                }
+                            }
+                        },
+                        "listeners": {
+                            "additionalProperties": {
+                                "oneOf": [
+                                    { "type": "string" },
+                                    "{gpii.schema.component}.options.schema.definitions.singleListenerDefinition",
+                                    {
+                                        "type": "array",
+                                        "items": "{gpii.schema.component}.options.schema.definitions.singleListenerDefinition"
+                                    }
+                                ]
+                            }
+                        },
+                        "members": { "type": "object"},
+                        "mergePolicy": {
+                            "type": "object",
+                            "patternProperties": {
+                                "^[a-z]+$": {
+                                }
+                            }
+                        },
+                        "schema": { "$ref": "gss-v7-full#"},
+                        "selectors": {
+                            "type": "object",
+                            "additionalProperties": { "type": "string" }
+                        },
+                        /*
+
+                        workflows: {
+                            local: {
+                                concludeComponentObservation: {
+                                    funcName: "fluid.concludeComponentObservation",
+                                    priority: "first"
+                                },
+                                concludeComponentInit: {
+                                    funcName: "fluid.concludeComponentInit",
+                                    priority: "last"
+                                }
+                            }
+                        }
+                         */
+                        // TODO: Talk with Antranig about how to model workflows correctly.
+                        "workflows": {
+                            "type": "object"
+                        }
+                    }
+                }
+            }
+        },
+        workflows: {
```

Comment:
We should factor this grade at this point so that we can somewhat gracefully support pre-6148 versions of Infusion, probably by positioning validation as a createOnEvent with priority: "first:authoring" listener. This may not be terribly satisfactory, but we should at least do something to actively support old Infusions since so far only gpii-universal and gpii-windows have been validated against the new branch and we may need to deploy this work before we can check out and fix gpii-app. At the very least, we should have a "sniff" in the tests which just skips the validated component checks if we find we're on an old Infusion. The best way of performing the sniff is probably to check whether fluid.registerPotentia is defined.

Reply:
I'm thinking of handling this with context awareness and "mix in" grades for pre-6148 and post.  These would be clearly labeled as non-API and not documented outside of the code itself.  I could use the same "check" in both the tests and the components.

The one thing I could use help with is figuring out how to test both pre-6148 and post.  I guess I could make a fake sub-package with a pre-6148 version of infusion and run the package tests from a wrapper file that requires the pre-6148 version there?  Or would that fall afoul of the deduping?